packages/gui/components/Select/Dropdown.tsx

```diff
@@ -18,6 +20,17 @@ const Dropdown: FC<Props> = ({ config }) => {
 	const { selected, items, title, getRequiredFields, onSelect } =
 		config.context as SelectionContext<object>;
 
+	const [searchText, setSearchText] = useState('');
+
+	const debouncedSearchText = useDebounce(searchText, 600);
+
+	const renderedItems = items.filter((item) => {
```

Comment:
with lodash `throttle` or `debounce`, that debounced method just internally `setSearchText` like normal.

this place should be something like:
```typescript
const filteredItems = useMemo(() => {
  return items.filter(() => {
     // searchText always up-to-date inside this memo, use it to filter out data
  });
}, [searchText]);
```